src/libraries/System.Private.CoreLib/src/System/DateOnly.cs

```
@@ -0,0 +1,902 @@
+        /// </summary>
+        /// <param name="s">The memory span that contains the string to parse.</param>
+        /// <returns>An object that is equivalent to the date contained in s.</returns>
+        public static DateOnly Parse(ReadOnlySpan<char> s) => Parse(s, DateTimeFormatInfo.CurrentInfo, DateTimeStyles.None);
+
+
+        private const ParseFlags ParseFlagsDateMask = ParseFlags.HaveHour | ParseFlags.HaveMinute | ParseFlags.HaveSecond | ParseFlags.HaveTime | ParseFlags.TimeZoneUsed |
+                                                      ParseFlags.TimeZoneUtc | ParseFlags.CaptureOffset | ParseFlags.UtcSortPattern;
+
+        /// <summary>
+        /// Converts a memory span that contains string representation of a date to its DateOnly equivalent by using culture-specific format information and a formatting style.
+        /// </summary>
+        /// <param name="s">The memory span that contains the string to parse.</param>
+        /// <param name="provider">An object that supplies culture-specific format information about s.</param>
+        /// <param name="style">A bitwise combination of enumeration values that indicates the permitted format of s. A typical value to specify is None.</param>
+        /// <returns>An object that is equivalent to the date contained in s, as specified by provider and styles.</returns>
+        public static DateOnly Parse(ReadOnlySpan<char> s, IFormatProvider? provider, DateTimeStyles style = DateTimeStyles.None)
+        {
+            if ((style & ~DateTimeStyles.AllowWhiteSpaces) != 0)
+            {
+                throw new ArgumentException(SR.Argument_InvalidDateStyles, nameof(style));
+            }
+
+            DateTimeResult result = default;       // The buffer to store the parsing result.
+            result.Init(s);
+
+            if (!DateTimeParse.TryParse(s, DateTimeFormatInfo.GetInstance(provider), style, ref result))
```

If the style is not within the range of accepted values what would happen? We would throw a format exception? Should we instead throw an argument exception?

---

Ah sorry I somehow forgot you are checking that just above 🤦